File: app/flytec/lib/features/aplications_v2/models/caracteristicas_produto_aplicado.dart
import 'dart:typed_data';

import 'package:flytec/features/produto/data/models/produto_model.dart';

class CaracteristicasProdutoAplicado {
  String? cultura;
  Uint8List? receiturarioAgronomico;
  String? nomeProduto;
  String? classificacaoToxicologica;
  String? classe;
  String? tipoFormulacao;
  String? alvoBiologico;
  String? doseProdutoHectare;
  String? unidadeDoseProdutoHectare;
  String? adjuvante;
  String? tipoServico;

  CaracteristicasProdutoAplicado({
    this.cultura,
    this.receiturarioAgronomico,
    this.nomeProduto,
    this.classificacaoToxicologica,
    this.classe,
    this.tipoFormulacao,
    this.alvoBiologico,
    this.doseProdutoHectare,
    this.unidadeDoseProdutoHectare,
    this.adjuvante,
    this.tipoServico,
  });

  Map<String, dynamic> toMap() {
    return {
      'cultura': cultura,
      'receiturarioAgronomico': receiturarioAgronomico,
      'nomeProduto': nomeProduto,
      'classificacaoToxicologica': classificacaoToxicologica,
      'classe': classe,
      'tipoFormulacao': tipoFormulacao,
      'alvoBiologico': alvoBiologico,
      'doseProdutoHectare': doseProdutoHectare,
      'unidadeDoseProdutoHectare': unidadeDoseProdutoHectare,
      'adjuvante': adjuvante,
      'tipoServico': tipoServico,
    };
  }

  factory CaracteristicasProdutoAplicado.fromJson(Map<String, dynamic>? json) {
    if(json==null) return CaracteristicasProdutoAplicado();
    return CaracteristicasProdutoAplicado(
      cultura: json['cultura'],
      receiturarioAgronomico: json['receiturarioAgronomico'],
      nomeProduto: json['nomeProduto'],
      classificacaoToxicologica: json['classificacaoToxicologica'],
      classe: json['classe'],
      tipoFormulacao: json['tipoFormulacao'],
      alvoBiologico: json['alvoBiologico'],
      doseProdutoHectare: json['doseProdutoHectare'],
      unidadeDoseProdutoHectare: json['unidadeDoseProdutoHectare'],
      adjuvante: json['adjuvante'],
      tipoServico: json['tipoServico'],
    );
  }

  factory CaracteristicasProdutoAplicado.fromProduto(ProdutoModel produto) =>
      CaracteristicasProdutoAplicado(
        nomeProduto: produto.nome,
        classificacaoToxicologica: produto.classificacaoToxicologica,
        classe: produto.classe,
        tipoFormulacao: produto.tipoDeFormulacao,
        tipoServico: produto.tipoServico,
      );
}

File: backoffice/Domain/Entidades/Cadastros/CaracteristicasProdutoAplicado/CaracteristicasProdutoAplicado.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Domain.Entidades.Cadastros.CaracteristicasProdutoAplicado;

public class CaracteristicasProdutoAplicado
{
    [Key]
    public int Id { get; set; }
    public string Cultura { get; set; }
    public string ReceiturarioAgronomico { get; set; }
    public string NomeProduto { get; set; }
    public int? ClassificacaoToxicologica { get; set; }
    public string Classe { get; set; }
    public string TipoFormulacao { get; set; }
    public string AlvoBiologico { get; set; }
    public string DoseProdutoHectare { get; set; }
    public string UnidadeDoseProdutoHectare { get; set; }
    public string Adjuvante { get; set; }
    public string TipoServico { get; set; }
    public string NumeroReceituarioAgronomico { get; set; }
    public string DataEmissao { get; set; }
    public bool? IsReceituarioImage { get; set; }

    [ForeignKey("Empresa")]
    public int? IdEmpresa { get; set; }

    [JsonIgnore]
    public virtual Empresa.Empresa? Empresa { get; set; }
}
